packages/environment-check/src/archive/index.ts

```diff
@@ -0,0 +1,75 @@
+import { createWriteStream, existsSync, promises } from 'fs';
+import { basename, dirname, join } from 'path';
+import * as archiver from 'archiver';
+import { glob } from 'glob-gitignore';
+import ignore from 'ignore';
+import { t } from '../i18n';
+import { byteNumberToSizeString } from '../formatter';
+
+/**
+ * Archive a project to zip file. Result file is written to parent of the project root folder.
+ *
+ * @param projectRoot - root of the project, where package.json is located
+ * @param targetFileName - optional file name, defaults to project folder + timestamp + .zip
+ */
+export async function archiveProject(
+    projectRoot: string,
+    targetFileName?: string
+): Promise<{ path: string; size: string }> {
+    if (!existsSync(projectRoot)) {
+        return Promise.reject(new Error(t('error.noProjectRoot', { projectRoot })));
+    }
+    const fileList = await getFileList(projectRoot);
+    return new Promise((resolve, reject) => {
+        try {
+            const zip = archiver.default('zip', { zlib: { level: 9 } });
+            let targetName = '';
+            if (typeof targetFileName === 'string') {
+                targetName = targetFileName.toLocaleLowerCase().endsWith('.zip')
+                    ? targetFileName
+                    : targetFileName + '.zip';
+            } else {
+                targetName = `${basename(projectRoot)}-${new Date()
+                    .toISOString()
+                    .replace('T', '')
+                    .replace(':', '')
+                    .substring(0, 14)}.zip`;
+            }
+            const targetPath = join(dirname(projectRoot), targetName);
+            const writeStream = createWriteStream(targetPath);
+            zip.pipe(writeStream);
+            zip.on('error', (error) => reject(error));
+            for (const file of fileList) {
+                zip.file(join(projectRoot, file), { name: file });
+            }
+            writeStream.on('close', () => resolve({ path: targetPath, size: byteNumberToSizeString(zip.pointer()) }));
+            zip.finalize();
+        } catch (error) {
+            reject(error);
+        }
+    });
+}
+
+/**
+ * Get list of files to be added to archive. Depending on the existence of file {cwd}/.gitignore, we use either a default filter
+ * or the .gitignore content.
+ *
+ * @param cwd - working directory, usually the project root
+ * @returns - list of files to add to the archive, relative to cwd
+ */
+async function getFileList(cwd: string): Promise<string[]> {
+    const gitignorePath = join(cwd, '.gitignore');
+    const hasGitignore = existsSync(gitignorePath);
+    const globPattern = hasGitignore ? ['**'] : ['**', '.cdsrc.json', '.extconfig.json'];
```

Comment:
thanks for including `.extconfig.json`

Reply:
> Not sure if `.env` should always be added to the ignore in case it is missing in the .gitignore file?

Hi @devinea, thanks for looking into this. When no `.gitignore` is present, all dot-files are excluded, so `.env` is also excluded by default. We only include the well known dot files for SAP projects, currently `.cdsrc.json` and `.extconfig.json`.